lib/router/router.js

```diff
@@ -26,37 +28,31 @@ export default class Router {
     }
   }
 
-  onPopState (e) {
+  async onPopState (e) {
     this.abortComponentLoad()
 
     const as = getURL()
     const { url = as } = e.state || {}
     const { pathname, query } = parse(url, true)
     const route = toRoute(pathname)
 
-    Promise.resolve()
-    .then(async () => {
-      const data = await this.fetchComponent(route)
-      const ctx = { ...data.ctx, pathname, query }
-      const props = await this.getInitialProps(data.Component, ctx)
-
-      this.route = route
-      this.set(pathname, query, { ...data, props })
-    })
-    .catch(async (err) => {
-      if (err.cancelled) return
-
-      const data = { Component: this.ErrorComponent, ctx: { err } }
-      const ctx = { ...data.ctx, pathname, query }
-      const props = await this.getInitialProps(data.Component, ctx)
+    this.emit('routeChangeStart', route, url)
```

Comment:
I'd like to pass both values user enter via `push()` or `replace()`.
I'll check them again.